formatter_replacer.go

```diff
@@ -156,7 +158,9 @@ func (r *fieldReplacer) format(key, value string, printKey bool, level logrus.Le
 		value += " "
 	}
 
+	r.outMutex.RLock()
 	sprint := r.out[level]
```

Comment:
We may use a sinc.Map instead.

Reply:
RWMutex is faster here since most operations are read and sync.Map will lock on one read while RWMutex allows unlimited concurrent reads

Also type safety is fun